Bottom: d6ef2da464caa0c937e4ca2fb4e7b1918d0d59f9
Top:    9a0cafea24d8de6191026ecee6cd2f1e5bdf07fc
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-01-07 01:00:08 +0000

BORAST: Fixup to only draw the passed POLYAREA, not walk the linked list of POLYAREA and draw them all!


---

diff --git a/src/borast/borast-bentley-ottmann.c b/src/borast/borast-bentley-ottmann.c
index aedbb37..4122b66 100644
--- a/src/borast/borast-bentley-ottmann.c
+++ b/src/borast/borast-bentley-ottmann.c
@@ -1383,6 +1383,7 @@ poly_area_to_start_events (POLYAREA                *poly,
                                  counter, outer_contour);
         outer_contour = 0;
       }
+      break;
 
     } while ((pa = pa->f) != poly);
 
@@ -1409,6 +1410,7 @@ bo_poly_to_traps (hidGC gc, POLYAREA *poly, borast_traps_t *traps)
     for (contour = pa->contours; contour != NULL; contour = contour->next)
       num_events += contour->Count;
     /* FIXME: Remove horizontal edges? */
+    break;
   } while ((pa = pa->f) != poly);
 
   if (unlikely (0 == num_events))
